src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/controller/PrisonerContactResource.kt

```diff
@@ -57,18 +48,40 @@ class PrisonerContactRegistryResource(
       ),
       ApiResponse(
         responseCode = "403",
-        description = "Incorrect permissions retrieve a Prisoner Contact",
+        description = "Incorrect permissions retrieve a Prisoner Contacts",
         content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
       ),
       ApiResponse(
         responseCode = "404",
-        description = "Prisoner Contact not found",
+        description = "Prisoner not found",
         content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
       ),
     ]
   )
   fun getPrisonerContact(
-    @Schema(description = "NOMIS Person Identifier", example = "ID12345", required = true)
-    @PathVariable nomisPersonId: String
-  ): ContactDTO = contactService.getContactById(nomisPersonId)
+    @Schema(description = "Prisoner Identifier (NOMIS Offender No)", example = "A1234AA", required = true)
+    @PathVariable prisonerId: String,
+    @RequestParam(value = "type", required = false)
+    @Parameter(
+      description = "Query by Type (NOMIS Contact Type)",
+      example = "S"
+    ) contactType: String?,
+    @RequestParam(value = "id", required = false)
+    @Parameter(
+      description = "Query by Person Identifier (NOMIS Person ID)",
+      example = "9147510"
+    ) personId: Long?
+  ): List<Contact> {
+    log.debug("Prisoner: $prisonerId, Type: $contactType, Person: $personId")
+    return contactService.getContactList(prisonerId, contactType, personId)
+
+    // 400 invalid prisonerId - bad request
```

Comment:
I assume this is a TODO to add these codes to the API docs?

Reply:
Decision notes of status codes, should have been removed - I will update.